internal/commands/ostest/test_execution.go

```diff
@@ -42,7 +46,54 @@ func RunTest(
 	errFactory *errors.ErrorFactory,
 	logger *zerolog.Logger,
 	localPolicy *testapi.LocalPolicy,
-) (*definitions.LegacyVulnerabilityResponse, workflow.Data, error) {
+) (*definitions.LegacyVulnerabilityResponse, []workflow.Data, error) {
+	finalResult, findingsData, err := executeTest(ctx, testClient, orgID, subject, localPolicy, errFactory, logger)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	// path should be the current working directory
+	currentDir, wdErr := os.Getwd()
+	if wdErr != nil {
+		logger.Error().Err(wdErr).Msg("Failed to get current working directory")
+		return nil, nil, fmt.Errorf("failed to get current working directory: %w", wdErr)
+	}
+
+	uniqueCount := calculateUniqueIssueCount(findingsData)
+
+	// The summary is always needed for the exit code calculation.
+	standardSummary, summaryData, summaryErr := NewSummaryData(finalResult, logger, currentDir)
+	if summaryErr != nil && !std_errors.Is(summaryErr, ErrNoSummaryData) {
+		// Log the error but continue, as this is not fatal.
+		logger.Warn().Err(summaryErr).Msg("Failed to create test summary for exit code handling")
+	}
```

Comment:
Exit code 1 should be determined by the test outcome from the API... so I'm not sure about this comment.